addons/binding/org.openhab.binding.konnected/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="konnected"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Sample Thing Type -->
+	<thing-type id="module">
+		<label>Konnected Binding Thing</label>
+		<description>The Konnected Module</description>
+		<channels>
+			<channel id="Zone_1" typeId="zone">
+				<label>Pin 1</label>
+			</channel>
+			<channel id="Zone_2" typeId="zone">
+				<label>Pin 2</label>
+			</channel>
+			<channel id="Zone_3" typeId="zone">
+				<label>Pin 3</label>
+			</channel>
+			<channel id="Zone_4" typeId="zone">
+				<label>Pin 4</label>
+			</channel>
+			<channel id="Zone_5" typeId="zone">
+				<label>Pin 5</label>
+			</channel>
+			<channel id="Zone_6" typeId="zone">
+				<label>Pin 6</label>
+				<description>Zone for Sensors Only</description>
+			</channel>
+			<channel id="Zone_7" typeId="actuator" />
+		</channels>
+
+		<config-description-ref uri="thing-type:konnected:module" />
+
+
+	</thing-type>
+
+	<!-- Zone Channel Type -->
+	<channel-type id="zone">
+		<item-type>String</item-type>
```

Comment:
Since they are read-only and the only valid states are "0" and "1" if i understand correctly, I would make this to a `Contact` item which has the two states `OPEN` and `CLOSED`.

Reply:
So my issue is that that zones 1-5 can either be actuators or sensors depending on what a user has hooked up to them.  When they are set as acutators the binding actively changes the description to allow them to be non read only. And in that instance the channel would be either off or on depending on the command issued.  I choose string type as a choice to allow the user to choose how they wanted it displayed in the UI.  Another potential problem is that in some instances a 0 can be off and with other senors it means on.    I know there is a way to dynamically change a develop channels at runtime and perhaps i need more deeply into that.  In that way a user who has defined zone 1 to be an actuator can have a swtich type and then contact types for zones 2-6 while another user could have contact types for zones 1-6. etc.

Reply:
Ok, now I see your problem. How about only offering channels which can be dynamically added to the thing?

You could offer the 2 types of channels, one for sending and one for receiving.

You can have a look at our magic bundle where we have a thing with extensible channels for testing:

https://github.com/eclipse/smarthome/blob/4c49e68e1142abb5506c470997165e8a2001127d/bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml#L14

Users can also easily add channels to their thing via paperui.